apps/site/src/lib/components/forms/MultipleSelect.tsx

```diff
@@ -0,0 +1,100 @@
+"use client";
+
+import { useState, useEffect, useRef, forwardRef } from "react";
+
+import RequiredAsterisk from "./RequiredAsterisk";
+
+interface MultipleInputs {
+	labelText: string;
+	identifierId: string;
+	values: Array<{ name: string; value: string; text: string }>;
+	containerClass: string;
+}
+
+interface OtherInputProps {
+	isChecked: boolean;
+	name: string;
+}
+
+const OtherInput = forwardRef<HTMLInputElement, OtherInputProps>(
+	({ isChecked, name }, ref) => (
+		<input
+			ref={ref}
+			type="text"
+			name={name}
+			className={
+				isChecked
+					? "text-[var(--color-black)] border-b-2 p-1 h-6 border-black w-6/12"
+					: "text-[var(--color-white)]border-b-2 p-1 h-6 border-black w-6/12 bg-transparent"
+			}
+			required={isChecked}
+			disabled={!isChecked}
+		/>
+	),
+);
+OtherInput.displayName = "OtherInput";
+
+export default function MultipleSelect({
+	labelText,
+	identifierId,
+	values,
+	containerClass,
+}: MultipleInputs) {
+	const [isOtherChecked, setIsOtherChecked] = useState(false);
+	const otherRef = useRef<HTMLInputElement>(null);
+
+	useEffect(() => {
+		if (isOtherChecked) {
+			otherRef.current?.focus();
+		}
+	}, [isOtherChecked]);
+
+	return (
+		<div className={containerClass}>
+			<p className="m-0 text-lg mb-4">
+				{labelText} <RequiredAsterisk />
+			</p>
+			<div className="w-10/12 flex flex-col gap-2">
+				{values.map((item, i) => {
+					const keyAndId = `${identifierId}-${i}`;
```

Comment:
Question: is the item name not sufficient as the key here?

Reply:
I don't know why this was actually done in [RadioSelect](https://github.com/HackAtUCI/irvinehacks-site/blob/main/apps/site/src/lib/components/forms/RadioSelect.tsx), but I changed it